test/integration/master/secrets_enveloping_test.go

```diff
@@ -79,117 +83,203 @@ resources:
         - name: key1
           secret: c2VjcmV0IGlzIHNlY3VyZQ==
 `
+
+	identityConfigYAML = `
+kind: EncryptionConfig
+apiVersion: v1
+resources:
+  - resources:
+    - secrets
+    providers:
+    - identity: {}
+`
 )
 
 type unSealSecret func(cipherText []byte, ctx value.Context, config encryptionconfig.ProviderConfig) ([]byte, error)
 
-// TestSecretsShouldBeEnveloped is an integration test between KubeAPI and ECTD that checks:
-// 1. Secrets are encrypted on write
-// 2. Secrets are decrypted on read
-// when EncryptionConfig is passed to KubeAPI server.
-func TestSecretsShouldBeEnveloped(t *testing.T) {
-	var testCases = []struct {
-		transformerConfigContent string
-		transformerPrefix        string
-		unSealFunc               unSealSecret
-	}{
-		{aesGCMConfigYAML, aesGCMPrefix, unSealWithGCMTransformer},
-		{aesCBCConfigYAML, aesCBCPrefix, unSealWithCBCTransformer},
-		// TODO: add secretbox
-	}
-	for _, tt := range testCases {
-		runEnvelopeTest(t, tt.unSealFunc, tt.transformerConfigContent, tt.transformerPrefix)
-	}
+type envelopTest struct {
+	logger            kubeapiservertesting.Logger
+	storageConfig     *storagebackend.Config
+	configDir         string
+	transformerConfig string
+	kubeAPIServer     kubeapiservertesting.TestServer
+	restClient        *kubernetes.Clientset
+	ns                *corev1.Namespace
+	secret            *corev1.Secret
 }
 
-func runEnvelopeTest(t *testing.T, unSealSecretFunc unSealSecret, transformerConfigYAML, expectedEnvelopePrefix string) {
-	transformerConfig := parseTransformerConfigOrDie(t, transformerConfigYAML)
+func newEnvelopeTest(l kubeapiservertesting.Logger, transformerConfigYAML string) (*envelopTest, error) {
+	var e envelopTest
+	e.logger = l
+	e.transformerConfig = transformerConfigYAML
+	e.storageConfig = framework.SharedEtcd()
+
+	var err error
+	if transformerConfigYAML != "" {
+		e.configDir, err = createKubeAPIServerEncryptionConfig(transformerConfigYAML)
+		if err != nil {
+			return nil, fmt.Errorf("error while creating KubeAPIServer encryption config: %v", err)
+		}
+	}
 
-	storageConfig := framework.SharedEtcd()
-	kubeAPIServer, err := startKubeApiWithEncryption(t, storageConfig, transformerConfigYAML)
+	e.kubeAPIServer, err = kubeapiservertesting.StartTestServer(l, e.getKubeAPIServerEncryptionOptions(), e.storageConfig)
 	if err != nil {
-		t.Error(err)
-		return
+		return nil, fmt.Errorf("failed to start KubeAPI server: %v", err)
 	}
-	defer kubeAPIServer.TearDownFn()
 
-	client, err := kubernetes.NewForConfig(kubeAPIServer.ClientConfig)
+	e.restClient, err = kubernetes.NewForConfig(e.kubeAPIServer.ClientConfig)
 	if err != nil {
-		t.Fatalf("error while creating client: %v", err)
+		return nil, fmt.Errorf("error while creating rest client: %v", err)
 	}
 
-	ns, err := createTestNamespace(client, testNamespace)
+	e.ns, err = createTestNamespace(e.restClient, testNamespace)
```

Comment:
all these assingments can be moved into the if, e.g.
```
if e.ns, err = createTestNamespace(e.restClient, testNamespace); err != nil {
  return nil, err
}
```

Reply:
Done.